server/src/test/java/org/elasticsearch/search/aggregations/metrics/cardinality/HyperLogLogPlusPlusTests.java

```diff
@@ -126,4 +133,66 @@ public void testPrecisionFromThreshold() {
         assertEquals(18, HyperLogLogPlusPlus.precisionFromThreshold(100000));
         assertEquals(18, HyperLogLogPlusPlus.precisionFromThreshold(1000000));
     }
+
+    private HyperLogLogPlusPlus makeRandomSketch(int size) {
+        // utility to make a random HLL of a given size
+        HyperLogLogPlusPlus counts = new HyperLogLogPlusPlus(MAX_PRECISION, BigArrays.NON_RECYCLING_INSTANCE, 0);
+        for (int i = 0; i < size; i++) {
+            final int n = randomInt(100_000_000);
+            final long hash = BitMixer.mix64(n);
+            counts.collect(0, hash);
+        }
+        return counts;
+    }
+
+    public void testReadWrite() {
+        // We'll test reading and writing HLL bytes for different cardinality sizes
+        // ranging from 1 to 10 million. Beyond 10M, it takes awhile to generate the
+        // random data.
+        int[] sizes = new int[] {
+            1,
+            10,
+            100,
+            1_000,
+            10_000,
+            100_000,
+            1_000_000, // 1 million
+            10_000_000 // 10 million
+        };
```

Comment:
Due to the above, this test takes about 1.5s to run, whereas other tests in this suite only take 500-700ms. May want to think about a way to make this test faster by using fewer unique values.